### PR TITLE
[8.2.x ] ISPN-4390 Realigned with ConcurrentHashMap from the JSR-166 repository

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-4390

- jsr166/src/jdk8/java/util/concurrent/ConcurrentHashMap.java 1.3
- factor out all inner classes/interfaces from BECHMV8
- refactor access to the Unsafe instace
- extract the package-level logic from ThreadLocalRandom

### CODE REVIEW

**Comment by tristantarrant** [comment]
@danberindei @wburns I've put the changes on this PR for 8.2.x only. I've factored out the inner classes as requested on the other PR, which I've closed

File: commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/Eviction.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package org.infinispan.commons.util.concurrent.jdk8backported;
+
+/**
+ * @author Will Burns
```
Comment (slaskawi) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/Eviction.java]: Do we consider this as a public API? If so, could you please add some Javadoc (or remove it completely if we don't).

File: commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/EvictionEntry.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.infinispan.commons.util.concurrent.jdk8backported;
+
+/**
+ * @author Will Burns
```
Comment (slaskawi) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/EvictionEntry.java]: Javadocs

File: commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/EvictionListener.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package org.infinispan.commons.util.concurrent.jdk8backported;
+
+import java.util.Map;
+
+/**
```
Comment (slaskawi) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/EvictionListener.java]: And here also

File: commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/EvictionPolicy.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
+package org.infinispan.commons.util.concurrent.jdk8backported;
+
+import java.util.Collection;
+
+/**
+ * // TODO: Document this
```
Comment (slaskawi) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/EvictionPolicy.java]: And here.

File: commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/SizeAndEvicting.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package org.infinispan.commons.util.concurrent.jdk8backported;
+
+/**
+ * @author Will Burns
```
Comment (slaskawi) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/SizeAndEvicting.java]: Javadocs

File: commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/Eviction.java (hunk 1 above)
Comment (tristantarrant) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/Eviction.java]: None of the classes in this package are considered public API

Comment (slaskawi) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/Eviction.java]: So maybe the javadocs should be simply removed... an empty thing is just rubbish...

**Comment by tristantarrant** [comment]
@slaskawi I've removed all the pointless javadocs

**Comment by slaskawi** [comment]
LGTM, integrating

**Comment by slaskawi** [comment]
Integrated, thanks Tristan! BTW - I updated fix versions for https://issues.jboss.org/browse/ISPN-4390

**Pull request merged**